### PR TITLE
fix(api): subscriber id being wrong id

### PR DESCRIPTION
**Pull request by davidsoderberg**
### What changed? Why was the change needed?
Our subscriber id is used to fetch preferences and not customers subscriber id.

### CODE REVIEW

File: apps/api/src/app/inbox/usecases/update-preferences/update-preferences.usecase.ts [hunk 1]
```diff
@@ -145,7 +145,7 @@ export class UpdatePreferences {
         channels: preference.channels,
         organizationId: command.organizationId,
         environmentId: command.environmentId,
-        subscriberId: command.subscriberId,
+        subscriberId: subscriber._id,
```
Comment (antonjoel82) [inline apps/api/src/app/inbox/usecases/update-preferences/update-preferences.usecase.ts:148]: **concern 👀**: We're not using this same `subscriber._id` in the `getSubscriberTemplatePreferenceUsecase` call. Why doesn't the command's `subscriberId` work?

**Review by antonjoel82 — COMMENTED** [review]
I will test now -- but confused why the `subscriber` ID is different than the the `command.subscriberId`

**Review by SokratisVidros — APPROVED** [review]

File: apps/api/src/app/inbox/usecases/update-preferences/update-preferences.usecase.ts (hunk 1 above)
Comment (davidsoderberg) [reply apps/api/src/app/inbox/usecases/update-preferences/update-preferences.usecase.ts:148]: Not sure I follow your concern here is that we do not use command.subscriberId? The commands subscriber id is the customer specified id not our mongo object id that is needed for queries in our database.

File: libs/application-generic/src/usecases/get-preferences/get-preferences.usecase.ts [hunk 2]
```diff
@@ -67,6 +67,11 @@ export class GetPreferences {
       .filter((preference) => preference !== undefined)
       .map((item) => item.preferences);
 
+    // ensure we don't merge on an empty list
+    if (preferences.length === 0) {
+      throw new NotFoundException('We could not find any preferences');
+    }
```
Comment (antonjoel82) [inline libs/application-generic/src/usecases/get-preferences/get-preferences.usecase.ts:73]: @davidsoderberg This prevents returning [] accidentally from this method

File: libs/application-generic/src/usecases/get-preferences/get-preferences.usecase.ts [hunk 3]
```diff
@@ -111,7 +116,7 @@ export class GetPreferences {
     return items.find(
       (item) =>
         item.type === PreferencesTypeEnum.SUBSCRIBER_WORKFLOW &&
-        item._templateId == templateId,
+        item._templateId === templateId,
```
Comment (antonjoel82) [inline libs/application-generic/src/usecases/get-preferences/get-preferences.usecase.ts:119]: 🗒 note (non-blocking): Typo fix‏

File: apps/api/src/app/inbox/usecases/update-preferences/update-preferences.usecase.ts [hunk 4]
```diff
@@ -176,7 +176,7 @@ export class UpdatePreferences {
       channels: preference.channels,
       organizationId: command.organizationId,
       environmentId: command.environmentId,
-      subscriberId: command.subscriberId,
+      subscriberId: subscriber._id,
```
Comment (BiswaViraj) [inline apps/api/src/app/inbox/usecases/update-preferences/update-preferences.usecase.ts:179]: I have made a similar change in my [PR here](https://github.com/novuhq/novu/pull/6475)
Can you please take a look at that

**Review by BiswaViraj — APPROVED** [review]

**Pull request merged**